tests/ui/never_type/diverging-place-match.rs

```diff
@@ -0,0 +1,15 @@
+//@ known-bug: #117288
+//@ check-pass
+
+#![feature(never_type)]
+
+fn make_up_a_value<T>() -> T {
+    unsafe {
+        let x: *const ! = 0 as _;
+        let _: ! = *x;
+        // Since `*x` "diverges" in HIR, but doesn't count as a read in MIR, this
+        // is unsound since we act as if it diverges but it doesn't.
+    }
```

Comment:
Can you make the return type of this function `!`? I think that makes it much clearer that we're testing for "HIR detecting divergence" here.